packages/@aws-cdk/aws-efs/lib/efs-file-system.ts

```diff
@@ -272,11 +272,13 @@ export class FileSystem extends Resource implements IFileSystem {
       defaultPort: ec2.Port.tcp(FileSystem.DEFAULT_PORT),
     });
 
+    this.mountTargetsAvailable = [];
+    if (props.vpcSubnets === undefined) return;
```

Comment:
Using `return` is dangerous here. Imagine this line changes location during some refactoring, it will now prevent all other logic in the constructor to run.

better to do:

```ts
if (props.vpcSubnets) {
  // configure mounts here
}
```